library/src/main/java/com/novoda/downloadmanager/LiteDownloadManagerDownloader.java

```diff
@@ -74,23 +74,21 @@ public void download(DownloadBatch downloadBatch, Map<DownloadBatchId, DownloadB
     private WaitForDownloadService.ThenPerform.Action<Void> executeDownload(DownloadBatch downloadBatch) {
         return () -> {
             InternalDownloadBatchStatus downloadBatchStatus = downloadBatch.status();
-            if (downloadBatchStatus.status() != DOWNLOADED) {
-                updateStatusToQueuedIfNeeded(downloadBatchStatus);
-                downloadService.download(downloadBatch, downloadBatchCallback());
-            }
+            updateStatusToQueuedIfNeeded(downloadBatchStatus);
+            downloadService.download(downloadBatch, downloadBatchCallback());
```

Comment:
You're calling this now even if its downloaded, I assume that's ok because it won't be queued and because there is a check in the `download` that will abort and invoke the callback immediately right?

Reply:
That's correct 😄